xscen/finalize.py

```diff
@@ -108,25 +116,56 @@ def clean_up(
         eg. {'global': {'title': 'amazing new dataset'}, 'tasmax': {'note': 'important info about tasmax'}}
     change_attr_prefix: str
         Replace "cat/" in the catalogue global attrs by this new string
+    to_level: str
+        The processing level to assign to the output.
 
     Returns
     -------
     ds: xr.Dataset
         Cleaned up dataset
     """
-    if var_and_convert_units:
-        logger.info(f"Converting units: {var_and_convert_units}")
-        ds = change_units(ds=ds, variables_and_units=var_and_convert_units)
+
+    if variables_and_units:
+        logger.info(f"Converting units: {variables_and_units}")
+        ds = change_units(ds=ds, variables_and_units=variables_and_units)
 
     # unstack nans
     if maybe_unstack_dict:
         ds = maybe_unstack(ds, **maybe_unstack_dict)
 
-    # put back feb 29th
-    if add_feb_29:
-        logging.info("Adding back february 29th by linear interpolation")
-        with_missing = convert_calendar(ds, "standard", missing=np.NaN)
-        ds = with_missing.interpolate_na("time", method="linear")
+    # convert calendar
+    if convert_calendar_kwargs:
+        ds_copy = ds.copy()
+
+        # if missing_by_var exist make sure missing data are added to time axis
+        if missing_by_var:
+            convert_calendar_kwargs["missing"] = np.nan
+
+        # make default `align_on`='`random` when the initial calendar is 360day
+        if get_calendar(ds) == "360_day" and "align_on" not in convert_calendar_kwargs:
+            convert_calendar_kwargs["align_on"] = "random"
+
+        logger.info(f"Converting calendar with {convert_calendar_kwargs} ")
+        ds = convert_calendar(ds, **convert_calendar_kwargs)
+
+        # convert each variable individually
+        if missing_by_var:
+            # remove 'missing' argument to be replace by `missing_by_var`
+            del convert_calendar_kwargs["missing"]
+            for var, missing in missing_by_var.items():
+                logging.info(f"Filling missing {var} with {missing}")
+                converted_var = convert_calendar(
+                    ds_copy[var], **convert_calendar_kwargs, missing=missing
+                )
+
+                if missing == "interpolate":
+                    converted_var = convert_calendar(
+                        ds_copy[var], **convert_calendar_kwargs, missing=np.nan
+                    )
+                    converted_var = converted_var.interpolate_na(
+                        "time", method="linear"
+                    )
```

Comment:
```suggestion
                if missing == "interpolate":
                    converted_var = convert_calendar(
                        ds_copy[var], **convert_calendar_kwargs, missing=np.nan
                    )
                    converted_var = converted_var.interpolate_na(
                        "time", method="linear"
                    )
                else:
                    converted_var = convert_calendar(
                        ds_copy[var], **convert_calendar_kwargs, missing=missing
                    )
```
Je sais que ça ne change pas grand chose, mais ça me faisait bizarre d'appeler `convert_calendar` avec `missing='interpolate'`...